product-management/src/scripts/controllers/user-controller.js

```diff
@@ -10,30 +10,75 @@ export default class UserController {
   initialize = () => {
     this.view.initialize();
     this.view.bindRegister(this.handleRegister);
-    // this.model.deleteUserById(4);
+    this.view.bindLogin(this.handleLogin);
   };
 
+  /**
+   * Pass the element and errorMessage to show the error signals including: 
+   * Hide the spinner 
+   * Show the invalid message 
+   * Show the red border
+   * @param {DOM} element 
+   * @param {String} errorMessage 
+   */
+  showError = (element, errorMessage) => {
+    this.view.hideElement(this.view.indexSpinner);
+    const invalidMessage = getInvalidMessageElement(element);
+    invalidMessage.innerHTML = errorMessage;
+    showInputError(element);
+  }
+
   /**
    * Handle registration 
    * Click submit to show spinner
-   * Check if email got from input field is existed in the database or not
-   * If it is existed, hide the spinner and show error signals and return 
-   * If it is a new one, show the spinner create a new user object with name, email, password information in the database
+   * Check if email got from input field exists in the database or not
+   * If it existed, hide the spinner and show error signals then return 
+   * If it is a new one, show the spinner and create a new user object with name, email, password information in the database
    * Then hide the spinner and redirect to the Products Page
    * @param {Object} user 
    */
   handleRegister = async (user) => {
     this.view.showFlexElement(this.view.indexSpinner);
-    if (await this.model.hasUser(user.email)) {
-      this.view.hideElement(this.view.indexSpinner);
-      const invalidMessage = getInvalidMessageElement(this.view.registerEmail);
-      invalidMessage.innerHTML = MESSAGES.EMAIL_EXISTED;
-      showInputError(this.view.registerEmail);
+    const hasUser = this.model.hasUser(user.email);
+    if (await hasUser) {
+      this.showError(this.view.registerEmail, MESSAGES.EMAIL_EXISTED);
       return;
     }
 
     await this.model.createNewUser(user);
     this.view.hideElement(this.view.indexSpinner);
     window.location.assign('./products.html');
   }
+
+  /**
+   * Handle Login 
+   * Click submit to show spinner
+   * Check if the email from input exists in the database or not
+   * If it doesn't, hide the spinner & show error signals
+   * If it does, compare the password got from database with the one from the input 
+   * If they match, hide the spinner & redirect to the Products page 
+   * If they don't, hide the spinner & show error signals to type the password again 
+   * @param {Object} user 
+   */
+  handleLogin = async (user) => {
+    this.view.showFlexElement(this.view.indexSpinner);
+    const hasUser = this.model.hasUser(user.email);
```

Comment:
Should create a class method to call this `this.model.hasUser(user.email);` in controller.